### PR TITLE
Autodownload tasks

### PR DESCRIPTION
**Pull request by hammoudhasan**
## Description

Describe your changes in detail.

I added automatic downloading of task files from Google Drive for code and ai society. Fixed https://github.com/camel-ai/camel/issues/156

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [x] Add automatic download of tasks for AI Society.
- [x] Add automatic download of tasks for Code.
- [x] Fix try except for Code SLURM part.

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: examples/ai_society/role_playing_multiprocess.py [hunk 1]
```diff
@@ -181,6 +183,42 @@ def main() -> None:
...
+        os.remove(zip_file_path)
+
+    # Remove any nested folder created during extraction
+    nested_folder_path = os.path.join(folder_path, "tasks")
+    if os.path.exists(nested_folder_path):
+        for file_name in os.listdir(nested_folder_path):
+            file_path = os.path.join(nested_folder_path, file_name)
+            destination_path = os.path.join(folder_path, file_name)
+            os.rename(file_path, destination_path)
+        os.rmdir(nested_folder_path)
```
Comment (lightaime) [inline examples/ai_society/role_playing_multiprocess.py:220]: I don't think we need to have two snippets of codes that do the same thing. Can we add it as a function to `utils`?

Comment (hammoudhasan) [reply examples/ai_society/role_playing_multiprocess.py:220]: Sure! On it.

Comment (hammoudhasan) [reply examples/ai_society/role_playing_multiprocess.py]: @lightaime please take a look now this must solve the issue

File: camel/utils.py [hunk 2]
```diff
@@ -186,3 +188,35 @@ def get_first_int(string: str) -> Optional[int]:
         return int(match.group())
     else:
         return None
+
+
+def download_tasks(task: TaskType, folder_path: str):
```
Comment (lightaime) [inline camel/utils.py:193]:
```suggestion
def download_tasks(task: TaskType, folder_path: str) -> None:
```

**Review by lightaime — APPROVED** [review]
Looks great. Left one small comment

File: camel/utils.py (hunk 2 above)
Comment (hammoudhasan) [reply camel/utils.py]: Addressed

**Comment by hammoudhasan** [comment]
I have changed the Google Drive link to Hugging Face link. Please take a look @lightaime.

**Review by lightaime — APPROVED** [review]
Perfect

**Pull request merged**